proto/storage/auth_machine_to_machine.proto

```diff
@@ -15,6 +15,7 @@ message AuthMachineToMachineConfig {
   enum Type {
     GENERIC = 0;
     GITHUB_ACTIONS = 1;
+    KUBE_SERVICE_ACCOUNT = 2;
```

Comment:
Should this be "ArgoCD" to be in line with github_actions? What does github actions mean (not familiar with this auth config)?

Reply:
The machine to machine (m2m) endpoint takes an identity token issued by an OIDC provider, validates it (based on m2m providers configured in Central), and issues a short-lived API token. This API token has roles based on the role mapping defined in the respective m2m configuration.

The type field is to represent different providers.  For the Github case, this enables a lot of built-in claim mappings OOTB.  I added a different type for Kube service accounts because in that case I need to add the HTTP `Authorization` header when fetching the OIDC configuration document and JWKS store.

Argo CD utilizes ACS CRs, which are used by the config-controller, which uses its K8s service account token to access the Central API.